src/ieee80211_station.erl

```diff
@@ -149,16 +157,23 @@ init([AC, ClientMAC, StationCfg = #station_config{bss = RadioMAC}]) ->
     Data = Data0#data{ac = AC,
 		      ac_monitor = ACMonitor,
 		      mac = ClientMAC},
-    {ok, initial_state(Data), Data}.
-
+    Data1 = update_rssi(FrameInfo, Data),
+    {ok, initial_state(Data1), Data1}.
+
+handle_event(Type, {'802.11', FrameType, DA, SA, BSS, FromDS, ToDS, Frame, FrameInfo}, State,
```

Comment:
I have to think about a different solution a bit more. But I can already tell you, this change is not acceptable, **never**, **ever**, **ever**. :rage:

Reply:
@RoadRunnr i can change every handle_event to catch FrameInfo without wrapping handle_event

Reply:
@platinumthinker thanks for all the changes! I gave it a quick try in ergw-capwap-node. `/api/v2/station` worked fine. However wtp-id looked pretty messed up -> `"wtp-id":"���w�7"`. Do I need to configure this separately somewhere?
Using this in a delete request to /`api/v2/wtp/���w�7/station/STATION_MAC` showed no result. Appending or deleting characters from the `WTP_ID` caused no error messages. Setting a random ID caused a message like -> `{"not_found":{"wtp_id":"01234"}}`. I guess this might be related to the strange ID.

The delete command in the version one API seems to work somewhat. I am able do remove  a device from the listing in the API. The data connection also breaks however the device stays associated to the WTP. Right now I tested this with one WTP only. I might repeat this and see if it makes a difference with multiple WTPs.

Reply:
@Florz, Hello! What wtp_id you have? with unicode symbols?

Reply:
@platinumthinker yes, I saw exactly the one I pasted here.

Reply:
@Florz, i mean what name expected? I tested with one wtp with name "wtp-lede" and all works fine. To fix faster this problem i need to repeat you environment..

Reply:
@platinumthinker that depends on which name you are providing. On my WTP inside the wtp.conf I specified `name = "wtp-ap";`.  The hostname is `LEDE` and inside the WTP certificate I am using `BIC-IRAP-WTP-03`.

/v1/wtp returns:
`[
  {
    "id": "���w�7",
    "endpoint": {
      "ip": "none",
      "port": "none"
    }
  },
  {
    "id": "BIC-IRAP-WTP-03",
    "endpoint": {
      "ip": "192.168.1.3",
      "port": 49416
    }
  },
  {
    "id": "���w�8",
    "endpoint": {
      "ip": "none",
      "port": "none"
    }
  }
]`
So this looks like `wtp_id` should be the name from the certificate and you might just be off by one in the data structure. The IDs of the disconnected APs seem to be sort of random but the first one looks familiar.
Using the certificate names is also a good idea because they have to be unique.

Reply:
@platinumthinker have you been able to reproduce this problem?

Reply:
@Florz, i have wtp router and pc with capwap-ac but i can't reproduce this problem. What "id" you excpect?

Reply:
@Florz can you get info about values from certificate using this wtp?

Reply:
@platinumthinker I think part of the problem is that I have only one of three WTPs connected. Even in the old API (see previous post), the non connected WTPs are listed with faulty IDs. The connected WTP provides a correct ID `BIC-IRAP-WTP-03`.
I think those disconnected WTPs should not be listed here. They probably show up because I configured them in `/etc/ergw-capwap-node/ergw-capwap-node.config`. In the config they are named accordingly as `<<"BIC-IRAP-WTP-01">>` and `<<"BIC-IRAP-WTP-02">>`.

The big problem now is that our STA is listed under the wrong WTP. As shown before the `wtp-id` is non printable and probably matches one of the non connected WTPs, whereas it should show up under WTP number 3.

Reply:
@Florz can you provide ergw-capwap-node.config?

Reply:
@platinumthinker see [here](https://pastebin.com/WwynQkdB)

Reply:
@Florz you have 3 wtp? and what you get from http api request "/api/v1/wtp"?
Another thing: what you have in logs?

Reply:
I have some config with several wtp (https://pastebin.com/SG5AqVmg) and not one wtp shows by commands api v1/v2..

Reply:
@platinumthinker I have three WTPs configured in `ergw-capwap-node.config` but only one is connected. The other two WTPs are not connected yet,

```
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v1/wtp"
[{"id":"���w�7","endpoint":{"ip":"none","port":"none"}},{"id":"BIC-IRAP-WTP-03","endpoint":{"ip":"192.168.1.3","port":51765}},{"id":"���w�8","endpoint":{"ip":"none","port":"none"}}]
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v1/station"
[{"id":"���w�7","address":{"ip":"none","port":"none"},"macs":["08-C5-E1-71-94-39"]}]
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v2/station"
[[{"wtp-id":"���w�7","radios":[{"radio-id":1,"stations":[{"mac":"08:c5:e1:71:94:39","rssi":-62,"vlan":0,"rx-packets":122,"rx-bytes":27268,"tx-packets":15,"tx-bytes":9098}]}]}]]
```
The phone is connected to `BIC-IRAP-WTP-03` for sure, as this is our only WTP at the moment. I uploaded my logs [here](https://pastebin.com/yuByvjJh).
I am also in contact with Andrey Teplyashin now, about this possible bug.

Reply:
@platinumthinker it becomes even stranger, when I connect the missing WTPs.

with `BIC-IRAP-WTP-02` added:
```
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v1/wtp"
[{"id":"���w�7","endpoint":{"ip":"none","port":"none"}},{"id":"BIC-IRAP-WTP-02","endpoint":{"ip":"192.168.1.2","port":56725}},{"id":"BIC-IRAP-WTP-03","endpoint":{"ip":"192.168.1.3","port":48080}},{"id":"���w�S","endpoint":{"ip":"none","port":"none"}},{"id":"���w�T","endpoint":{"ip":"none","port":"none"}},{"id":"���w�8","endpoint":{"ip":"none","port":"none"}}]

user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v2/station"
[[{"wtp-id":"���w�7","radios":[{"radio-id":1,"stations":[{"mac":"08:c5:e1:71:94:39","rssi":-60,"vlan":0,"rx-packets":41,"rx-bytes":8357,"tx-packets":8,"tx-bytes":3956}]}]}]]
```
with all WTPs attached:
```
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v1/wtp"
[{"id":"���w�\u0004","endpoint":{"ip":"none","port":"none"}},{"id":"���w�7","endpoint":{"ip":"none","port":"none"}},{"id":"BIC-IRAP-WTP-02","endpoint":{"ip":"192.168.1.2","port":59793}},{"id":"BIC-IRAP-WTP-03","endpoint":{"ip":"192.168.1.3","port":51795}},{"id":"���w�S","endpoint":{"ip":"none","port":"none"}},{"id":"���w�T","endpoint":{"ip":"none","port":"none"}},{"id":"���w�8","endpoint":{"ip":"none","port":"none"}},{"id":"BIC-IRAP-WTP-01","endpoint":{"ip":"192.168.1.1","port":39676}},{"id":"���w�\u0003","endpoint":{"ip":"none","port":"none"}}]
```
in this case it also messes up the v2 station list:
```
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v1/station"
[{"id":"BIC-IRAP-WTP-02","address":{"ip":"192.168.1.2","port":59793},"macs":["08-C5-E1-71-94-39"]}]
user@IRAP:~$ curl -X "GET" "http://127.0.0.1:8000/api/v2/station"
[[{"wtp-id":"���w�7","radios":[]}]]
```

Reply:
fixed this problem in last commit (https://github.com/travelping/capwap/pull/25/commits/adc4fd758208f6a05937c21c9b7aa4d6bcd98f64)